src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -232,7 +232,7 @@ class MoneyRequestAmountPage extends React.Component {
      * @param {String} key
      */
     updateAmountNumberPad(key) {
-        if (!this.textInput.isFocused()) {
+        if (this.state.shouldUpdateSelection && !this.textInput.isFocused()) {
```

Comment:
Nice that you have incorporated the suggestion as well https://github.com/Expensify/App/issues/19226#issuecomment-1557673254